doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,223 @@
+# HTTP Transport Adapter
+
+The Azure SDK for C++ creates and sends HTTP requests using an HTTP Transport Adapter. This adapter provides a general abstraction for an HTTP stack (like libcurl, WinHTTP, etc) by defining the interface it needs to call that anyone can implement to customize which HTTP stack gets used.
+
+## Default HTTP Transport Adapter
+
+The Azure SDK for C++ provides two HTTP transport adapters as part of the azure-core package:
+- The libcurl Transport Adapter
+
+<center>
```

Comment:
Is this html-like tag necessary? It's fine to keep if it is important, but I am curious what happens if you don't add that?

Reply:
This makes the table to be centered like:
![image](https://user-images.githubusercontent.com/24213737/104072749-e3a8e700-51c0-11eb-9e7a-d8d767ce9732.png)

instead of:
![image](https://user-images.githubusercontent.com/24213737/104072773-f0c5d600-51c0-11eb-9b1b-382f29066a16.png)

Reply:
That's not what I see here:
https://github.com/Azure/azure-sdk-for-cpp/blob/8238c6de7ab413461d79134f4baed3d4380c2654/doc/HttpTransportAdapter.md#default-http-transport-adapter
![image](https://user-images.githubusercontent.com/6527137/104073149-c7f21080-51c1-11eb-8bbd-5fbd5ec65bbb.png)

Reply:
oh that's a shame!  It looks like github doesn't support it.
Still think it's nice to have it for any customer opening the md file in some better reader which supports the tag

Reply:
The markdown linter doesn't like the use of raw HTML tags for the opposite reason. Some markdown readers don't work if you use non-pure elements:
https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md033---inline-html